packages/react-provider/src/index.tsx

```diff
@@ -122,11 +122,15 @@ export const CourierProvider: React.FunctionComponent<ICourierContext> = ({
       return;
     }
 
+    if (!graphQLClient.canRequest) {
+      return;
+    }
+
     dispatch({
       type: "root/GET_BRAND",
       payload: () => getBrand(graphQLClient, brandId),
     });
-  }, [brand, brandId]);
+  }, [graphQLClient, brand, brandId]);
```

Comment:
would `graphQLClient.canRequest` as deps give more control over when this effect is called?

Reply:
it `could` but it doesn't because this is how the client is created:

```
  const graphQLClient = useMemo(() => {
    return new Client({ clientKey, userId, userSignature, apiUrl });
  }, [clientKey, userId, userSignature, apiUrl]);
```

Reply:
so it makes a new one anyway